### PR TITLE
INK-36: created user model and its test cases

### PR DESCRIPTION
**Pull request by MahmoudSaeed13**
- Created the user model.
- Written the test cases to test all the parameters of every property of the user object. 
- Created conftest.py to be a module for all the fixtures in the test directory.
- Modified the admin_user fixture to pass the name parameter to create_superuser method as it is a required field.

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+logs:
+	docker-compose -f local.yml logs -f $(filter-out $@,$(MAKECMDGOALS))
+
+# TODO: fix command to handle custom command
+test:
+	docker-compose -f local.yml run --service-ports --rm -e DEBUGGER=True -e DJANGO_SETTINGS_MODULE=config.settings.test django pytest $(filter-out $@,$(MAKECMDGOALS))
+
+test_local:
+	docker-compose -f local.yml exec -e DJANGO_SETTINGS_MODULE=config.settings.test django /entrypoint python manage.py test $(filter-out $@,$(MAKECMDGOALS))
```
Comment (ahmedeltaweel) [inline makefile:44]: Take care of pytest as it's not like unittest.
and validate that we can send custom parameters

File: theinkspot/users/admin.py [hunk 2]
```diff
@@ -13,13 +13,23 @@ class UserAdmin(auth_admin.UserAdmin):
 
     form = UserAdminChangeForm
     add_form = UserAdminCreationForm
+    add_fieldsets = (
```
Comment (ahmedeltaweel) [inline theinkspot/users/admin.py:16]: Would you please attach screenshot ?

File: theinkspot/users/forms.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
 from allauth.account.forms import SignupForm
 from allauth.socialaccount.forms import SignupForm as SocialSignupForm
+from django import forms
```
Comment (ahmedeltaweel) [inline theinkspot/users/forms.py:3]: feel free to remove the forms file and its logic.As we will not use it we will rely on APIs

File: theinkspot/users/migrations/0001_initial.py [hunk 4]
```diff
@@ -1,6 +1,5 @@
-# Generated by Django 3.2.9 on 2021-11-20 11:23
-import django.contrib.auth.models
-import django.contrib.auth.validators
+# Generated by Django 3.2.12 on 2022-05-01 11:42
```
Comment (ahmedeltaweel) [inline theinkspot/users/migrations/0001_initial.py:1]: Please squash to only 1 migration per PR.

File: theinkspot/users/models.py [hunk 5]
```diff
@@ -1,26 +1,72 @@
...
 
-class User(AbstractUser):
-    """
-    Default custom user model for TheInkSpot.
-    If adding fields that need to be filled at user signup,
-    check forms.SignupForm and forms.SocialSignupForms accordingly.
-    """
+class UserManager(BaseUserManager):
+    def create_user(self, username, email, name, password=None):
+        if username is None:
```
Comment (ahmedeltaweel) [inline theinkspot/users/models.py:14]:
```suggestion
        if username:
```

**Review by ahmedeltaweel — CHANGES_REQUESTED** [review]
LGTM! some comments to consider, please check them.

File: theinkspot/users/admin.py (hunk 2 above)
Comment (MahmoudSaeed13) [reply theinkspot/users/admin.py:16]: Screenshot from 2022-05-06 20-10-32
Screenshot from 2022-05-06 20-10-19

we added the field (is user verified by email) to the user view page & add two fields(name & email) to the add user form

File: theinkspot/users/forms.py (hunk 3 above)
Comment (MahmoudSaeed13) [reply theinkspot/users/forms.py:3]: we will not be working with forms, we needed this only for the admin panel.

**Review by ahmedeltaweel — APPROVED** [review]
LGTM! only comment is to remove vscode config from code repo.
add .vscode to .gitignore

**Pull request merged**